app/core/voyagerUtils.js

```diff
@@ -0,0 +1,93 @@
+
+module.exports = {
+
+  /**
+   * Creates a level status map using the list of level session objects.
+   * @param {Object[]} sessions - The list of level session objects.
+   * @returns {Object} - Object with key as the level original id, and value as complete/started.
+   */
+  getLevelStatusMap: function (sessions) {
+    let levelStatusMap = {}
+    for (let session of sessions) {
+      let levelOriginal = (session.level || session.get('level') || {}).original
+      if (levelOriginal && levelStatusMap[levelOriginal] !== 'complete') { // Don't overwrite a complete session with an incomplete one
+        if ((session.state || session.get('state') || {}).complete) {
+          levelStatusMap[levelOriginal] = 'complete'
+        } else {
+          levelStatusMap[levelOriginal] = 'started'
+        }
+      }
+    }
+    return levelStatusMap
+  },
+
+  /**
+   * Creates a level data map using the list of level objects.
+   * @param {Object[]} levels - The list of level objects.
+   * @returns {Object} - Object with key as the level original id, and value as the object containing level's data.
+   */
+  getLevelDataMap: function (levels) {
+    let levelDataMap = {}
+    for (let level of levels) {
+      let levelOriginal = level.original || level.get('original')
+      if (levelOriginal) {
+        levelDataMap[levelOriginal] = level.attributes || level
+      }
+    }
+    return levelDataMap
+  },
+
+  /**
+   * Gets the next level original id for a given level
+   * @param {Object} level - The level object.
+   * @param {Object[]} level.nextLevels - The array of nextLevels for the given level.
+   * @returns {string|undefined} - 'Original' id of the next level, or undefined.
+   */
+  getNextLevelOriginalForLevel: function (level) {
```

Comment:
I agree there is currently only one level currently. However once there are multiple levels we will want to extend this function.
Should we instead return an array of strings as possible next levels?
Then when there are many next levels this logic won't change.

Reply:
Yeah that's a good point, will update. Thank you